enhancements/single-node/single-node-openshift-day2-workers.md

```diff
@@ -0,0 +1,290 @@
+
+
+### User Stories
+
+- As an OpenShift Single Node administrator, I want to add worker nodes to my
+  single-node cluster, so that it'll be able to handle growing computation demands.
+
+### API Extensions
+
+This enhancement does not modify/add any API
+
+### Implementation Details/Notes/Constraints
+
+This enhancement can be easily implemented by adjusting the installer's
+`generateDefaultIngressController` method such that when `config.platform.none
+!= nil`, it will return an `IngressController` CR struct with
+`.spec.nodePlacement.nodeSelector.matchLabels` targetting
+`node-role.kubernetes.io/master`. 
+
+### Risks and Mitigations
+
+This should make no noticable difference on "regular" single-node installations
+which do not go through expansion (as the default node selector targets the
+worker pool, and the single control plane is already both in the master and
+worker pools), but it will fix the issue described above for single-node
+installations which do go through expansion.
+
+Users who *want* their traffic to not go through the single control-plane node
+can still do so by following the [existing](https://docs.openShift.com/container-platform/4.9/machine_management/creating-infrastructure-machinesets.html#moving-resources-to-infrastructure-machinesets) OpenShift documentation on moving
+infrastructure workloads to particular nodes.
+
+I do not believe this enhancement has any security implications.
+
+## Design Details
+
+### Open Questions
+
+None that I can think of at the moment
+
+### Test Plan
+
+- Add unit tests to the installer to make sure the IngressController manifest
+  is generated as exepected.
+
+- Add periodic nightly tests which install single-node in the cloud, add
+  a few worker nodes to it, then run conformance tests to make sure we don't
+  run into any problems not described in this enhancement.
+
+- Add periodic nightly tests which install a single-node "none"-platform
+  cluster, add worker nodes to it, and check that ingress traffic still works as
+  expected and recovers even after the `router-default` pod gets deleted and
+  rescheduled. Make sure this is still true even after upgrades.
+
+- Add tests on both cloud / "none"-platform that check that a single-node
+  cluster with additional workers recovers after the single control-plane node
+  reboots by running conformance tests post-reboot.
+
+- Add tests on both cloud / "none"-platform that check that a single-node
+  cluster with additional workers recovers after an upgrade by running
+  conformance tests post-upgrade.
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+TBD
+
+#### Dev Preview -> Tech Preview
+
+TBD
+
+#### Tech Preview -> GA
+
+TBD
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+In the non-goals section it's mentioned that this enhancement does not apply to
+clusters which have been installed prior to the enhancement, so their upgrade
+is not discussed.
```

Comment:
Is there a way that existing clusters can modify their cluster to match the outcome of this enhancement?

Reply:
Yes, see note in non-goals

>(they can be taken care of with a simple documentation on how it can be
  enabled with a few manual steps)

Reply:
All they would have to do is adjust their IngressController from `worker` to `master` I believe

Reply:
Is it worth including a short example of how this can be done within the enhancement?